### ISSUE TITLE
[p5.js 2.0 Beta Bug Report]: FES support for preload

### ISSUE THREAD

**Issue by aferriss** [processing/p5.js#7540]
### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build process
- [ ] Unit testing
- [ ] Internationalization
- [x] Friendly errors
- [ ] Other (specify if possible)

### p5.js version

2.0.0 beta 2

### Web browser and version

Chrome 132.0.6834.111

### Operating system

MacOSX Sequoia 15.2

### Steps to reproduce this

I was wondering if there should be some FES message when trying to use a loading function within preload, or preload at all in 2.0. As it stands, things just fail silently and the preload function is never called. Maybe we can direct the user to some information about the new changes and how they might adapt their code to begin working again?

> In p5 2.0, preload is no longer supported. In order to load assets you need to use the async and await keywords in the setup functions or use callbacks..... here's a link to learn more about the changes

Apologies if this is a duplicate issue, I saw some others but they seemed a little different ( #7533)

### Steps:
1. Call any `loadX()` function in preload 
2. Observe nothing happens.

### Snippet:

```js
let img;

// preload never called
function preload(){
   img = loadImage("test.jpg");
}

function setup(){
  createCanvas(100, 100);
}

function draw(){
}

```

**Comment by ksen0** [comment]
Hi @aferriss thanks for suggesting this! Yes, this is related to [the other issue](https://github.com/processing/p5.js/issues/7533), but you're right, what you're describing is different. 

I agree that even if no errors are caused (such as by trying to use `img`) a proactive message by FES 2.0 for when code contains `preload` would be really helpful. The current timeline for documentation is to have 2.0 reference ready by the time of first release end of March 2025, so that's when a usable link could be available.

https://github.com/processing/p5.js/issues/7533 is about non-silent failure, and particularly errors that don't seem like they're obviously related to loading. It seems to me that if there is an error (such as by trying to use `img`) then there should only be one FES message (not sure if it should be the same message or something worded differently), so maybe the actual implementation for these two things should be combined?

Is this something you'd be interested in working on?

**Comment by aferriss** [comment]
@ksen0 Thanks for the quick reply! I don't think I have the time at the moment to work on this but I'm happy to keep beta testing or give advice to anyone else working on the issue.

**Comment by aferriss** [comment]
@ksen0 I made a draft PR here super quickly, there's probably more to be done but if anyone wants to pick this up and take it further maybe this is a good start!
https://github.com/processing/p5.js/pull/7542

**Comment by aferriss** [comment]
One more thing! Just doing a search for `preload()` in the repo turns up a zillion results. It seems like it might be a pretty big effort to change all the inline examples to no longer use that function. Just flagging in case it wasn't already on the roadmap!
